lib/auth/user_agent.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Auth.UserAgent do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Auth.Repo
+
+  schema "user_agents" do
+    field :user_agent, :string
+    has_many :login_logs, Auth.LoginLog
+  end
+
+  @doc false
+  def changeset(user_agent, attrs) do
+    user_agent
+    |> cast(attrs, [:user_agent])
+    |> validate_required([:user_agent])
+    |> unique_constraint(:user_agent)
+  end
+
+  def get_or_insert_user_agent(ua) do
+    case Repo.get_by(Auth.UserAgent, user_agent: ua) do
+      nil ->
+        changeset(%Auth.UserAgent{}, %{user_agent: ua})
+        |> Repo.insert!(on_conflict: :nothing)
```

Comment:
does the `unique_constraint(:user_agent)` enforced in the `changeset` remove the need for the `on_conflict: :nothing`?

Reply:
I'll do some more tests on this it was one of the option I wasn't 100% sure